### DISCUSSION TITLE
Distributed deployment - logs stuck in redis

### DISCUSSION THREAD

**Discussion by johnununu** [Security-Onion-Solutions/securityonion#4917] [Unsupported Versions] [answered]
Thank you for all of the hard work that goes into Security Onion.

I upgraded from 2.3.52 to 2.3.60, and now 2.3.61.

With the filebeat module upgrade in 2.3.60, I've enabled several (cisco ios, ftd, squid).  I can see the logs making their way from forwarding nodes to the manager and see them getting into Redis.  The logstash instances on the storage nodes do not seem to be able to pull the module logs down so nothing comes back in elasticsearch and of course nothing is displayed in kibana.

Has anyone else ran into this issue?

**Comment by defensivedepth** [comment]
Are you seeing any relevant logs in Logstash or Filebeat?

`/opt/so/log/logstash/logstash.log`
`/opt/so/log/filebeat/filebeat.log`

**Comment by johnununu** [comment]
Good morning Josh,

Thanks for reaching out.  I am not seeing the logs in logstash or filebeat on the storage nodes or the manager.
I do see the logs coming in on the Redis docker interface on the manager.
So the logs make it to Redis, but that’s it.  As far as I can tell.

**Comment by johnununu** [comment]
I see the docker logstash interface is getting the logs on the manager via “tcpdump -nnSs 0 -i veth…..”

So logstash and Redis on the manager do appear to be working as normal.

**Comment by defensivedepth** [comment]
`/opt/so/log/logstash/logstash.log` is where errors are logged for Logstash. Can you tail that and post anything that looks out of place?

Also - you are seeing other log types in Elasticsearch ie Zeek, Suricata, etc?

**Comment by johnununu** [comment]
There’s nothing that looks out of place to me.  I’m still able to view ossec/wazuh, zeek and typical syslog.

I wonder if it’s because I’m sending module compatible (ios,ftd,squid) logs to the forward nodes, letting filebeats process the logs and then sending the processed logs to the manager.

Is there something extra I need to do in elastic search to enable filebeat module indexes?
The only error I see there is that Legacy index templates are deprecated in favor of composable templates.

**Reply by defensivedepth** [reply]
Can you post the filebeat configuration that you added for these modules? (sanitized as needed)

**Comment by johnununu** [comment]
I added to the /opt/so/saltstack/local/pillar/minions/*_sensor.sls


firewall:
  assigned_hostgroups:
    chain:
      DOCKER-USER:
        hostgroups:
          proxies:
            portgroups:
              - portgroups.squid
          logging_server:
            portgroups:
              - portgroups.cisco_ftd
              - portgroups.cisco_ios

filebeat:
  third_party_filebeat:
    modules:
      cisco:
        ftd:
          enabled: true
          var.syslog_host: 0.0.0.0
          var.syslog_port: 9003
        ios:
          enabled: true
          var.syslog_host: 0.0.0.0
          var.syslog_port: 9002
      netflow:
        log:
          enabled: true
          var.netflow_host: 0.0.0.0
          var.netflow_port: 2055
          var.internal_networks:
            - private
      squid:
        log:
          enabled: true
          var.input: udp
          var.syslog_host: 0.0.0.0
          var.syslog_port: 9520

--------------------------------------------------------------------------------

A separate issue is that netflow isn’t listening, so I didn’t create a firewall rule for it.

salt "*_sensor" cmd.run "docker ps | grep so-fileb"

_sensor:
    0e166e445d68   nemionimp01:5000/security-onion-solutions/so-filebeat:2.3.61             "/usr/local/bin/dock…"   41 hours ago   Up 19 hours             0.0.0.0:514->514/tcp, 0.0.0.0:5066->5066/tcp, 0.0.0.0:514->514/udp, 0.0.0.0:9002-9003->9002-9003/tcp, 0.0.0.0:9002-9003->9002-9003/udp, 0.0.0.0:9520->9520/tcp, 0.0.0.0:9520->9520/udp   so-filebeat

**Comment by johnununu** [comment, marked as answer]
Ended up just redeploying to the updated version and the issue is resolved.
Please close.
